mentors/views.py

```diff
@@ -1,35 +1,30 @@
-# TODO: remove unused imports
-from rest_framework import status
-from rest_framework.decorators import api_view, permission_classes
-from rest_framework.permissions import IsAdminUser
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework.permissions import AllowAny
 from rest_framework.response import Response
+from utils.swagger import set_example
 from .models import Mentor
-from .serializers import MentorSerializer, MentorListSerializer
-from decorators import ecell_user
-from django.http import HttpResponse
-import csv
+from .serializers import MentorListSerializer
+from rest_framework import status, generics
+from . import responses
 
 
-# TODO: simplify with drf
-@api_view(['GET', ])
-def get_mentors(request, year):
+class MentorView(generics.ListAPIView):
+    permission_classes = [AllowAny, ]
+    queryset = Mentor.objects.all()
+    serializer_class = MentorListSerializer
 
-    res_message = ""
-    res_status = ""
-    res_data = []
-
-    mentors = Mentor.objects.filter(year=year, flag=True)
-    if len(mentors) > 0:
-        res_data = MentorListSerializer(
-            mentors, many=True, context={
-                'request': request}).data
-        res_message = "Mentors Fetched successfully."
-        res_status = status.HTTP_200_OK
-    else:
-        res_message = "Mentors Couldn't be fetched"
-        res_status = status.HTTP_404_NOT_FOUND
-
-    return Response({
-        "message": res_message,
-        "data": res_data
-    }, status=res_status)
+    @swagger_auto_schema(
+        operation_id='get_mentors',
+        responses={
+            '200': set_example(responses.get_mentors_200),
+            '404': set_example(responses.mentors_not_found_404),
+        },
+    )
+    def get(self, request, year):
+        queryset = Mentor.objects.filter(year=year, flag=True)
+        serializer = MentorListSerializer(queryset, many=True)
+        data = serializer.data
+        if queryset.count() > 0:
+            return Response({"message": "Mentors Fetched successfully.", "data": data}, status.HTTP_200_OK)
+        else:
+            return Response({"message": "Mentors Couldn't be fetched"}, status.HTTP_404_NOT_FOUND)
```

Comment:
Since this is the `else` body of the condition `if queryset.count() > 0:`, this actually signifies that `queryset.count()==0` (assuming `count` cannot be negative ofcourse).
Thereby it signifies that the Mentors were fetched, its just that there are no mentors. We can return an empty dictionary i.e. present in `data` in such a case. It does not mean that Mentors couldn't be fetched.

Reply:
Rightly pointed out @priyanshisharma 🙇

Reply:
Removed the pointed else body